helpers/controllers/authController.test.js

```diff
@@ -81,4 +89,75 @@ describe('auth controller', () => {
       success: true
     })
   })
+
+  test('Signup - should resolve with success false if userinfo is incomplete', async () => {
+    const result = await signup({}, {}, { req: { session: {} } })
+    expect(result).toEqual({
+      success: false,
+      error: 'Invalid registration information'
+    })
+  })
+
+  test('Signup - should resolve with success false if user already exists', async () => {
+    db.User.findOne = jest.fn().mockReturnValue({ username: 'c0d3user' })
+    const result = await signup({}, userArgs, { req: { session: {} } })
+    expect(result).toEqual({
+      success: false,
+      error: 'User already exists'
+    })
+  })
+
+  test('Signup - should resolve with success false if email already exists', async () => {
+    db.User.findOne = jest
```

Comment:
This test structure seems odd to me (this could be since I'm still new to all this testing stuff lol)
But like, this test case, for example, seems to depend on the one before it, and it's also mocking a function return value twice which at first glance throws me off until I look at the `signup` code.

And also, I feel like the mocked return value's don't really make sense to me. Like if I testing a function that should tell me if the email already exists shouldn't I mock something like `{ username: '...', email: '[expected email]' }`

Reply:
for this, I looked at the code. The first check was for username (which returned null), the second check (which is for email) returned an object (meaning user found). What you are saying would make sense if this function was a `getByEmail` function or something, so maybe the description could be better:

`Should resolve with false If second user lookup (which is email lookup) returns a user object, meaning a user was found by email.`